### PR TITLE
fix: [CDS-52387]: adding index on instance collection

### PR DESCRIPTION
**Pull request by lasdox**
## Describe your changes
Adding index on instance collection for better performance on query.
Also removing an offending hint

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44633)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java [hunk 1]
```diff
@@ -419,7 +419,6 @@ private List<Instance> getInstancesForAccount(String accountId, long timestamp,
           .field(InstanceKeys.deletedAt)
           .greaterThanOrEq(timestamp);
       FindOptions findOptions = wingsPersistence.analyticNodePreferenceOptions();
-      findOptions.hint(BasicDBUtils.getIndexObject(Instance.mongoIndexes(), "instance_index7"));
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java:422]: do we not need hint here?

**Review by abhinav-harness — CHANGES_REQUESTED** [review]

**Review by abhinavmittal-harness — COMMENTED** [review]
> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44633)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java` line 422 at r1](https://reviewable.io/reviews/harness/harness-core/44633#-NPaC5JzERyxzWtSRCVp:-NPaC5JzERyxzWtSRCVq:bk0cyd2) ([raw file](https://github.com/harness/harness-core/blob/5e6b8056a77cddd39710b713c32458f76f54fd05/400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java#L422)):*
> ```Java
>           .greaterThanOrEq(timestamp);
>       FindOptions findOptions = wingsPersistence.analyticNodePreferenceOptions();
>       findOptions.hint(BasicDBUtils.getIndexObject(Instance.mongoIndexes(), "instance_index7"));
> ```

We can add the new hint here as Abhinav said

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 3 files at r1, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44633#-:-NPdCPPl1vy0xoHpesYz:bnfp4nl)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by rafaelmart1ns — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44633)* status: :shipit: complete! ✓

**Review by franciscojuniorharness — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44633)* status: :shipit: complete! ✓

**Pull request merged**